flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/table/lookup/CachingAsyncLookupFunction.java

```diff
@@ -124,10 +125,10 @@ public LookupCache getCache() {
     }
 
     // --------------------------------- Helper functions ----------------------------
-    private void updateLatestLoadTime() {
+    private synchronized void updateLatestLoadTime(long loadTime) {
```

Comment:
``` 
        @Override
    public void close() throws Exception {
        delegate.close();
        if (cacheIdentifier != null) {
            LookupCacheManager.getInstance().unregisterCache(cacheIdentifier);
        }
    }
```
minor: check null before close

Reply:
Thanks for the review! The null check will be included in the full caching implementation. I'll leave it as is for now.